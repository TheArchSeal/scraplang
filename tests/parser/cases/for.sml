for (;;);
for (x;x;x)x;
for (var i = 0; i < 10; i++);
