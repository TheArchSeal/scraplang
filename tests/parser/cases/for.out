stmt (10):1:1 for
    stmt (1):1:6 (nop)
    expr (1):1:7 (empty)
    expr (1):1:8 (empty)
    stmt (1):1:9 (nop)
stmt (10):2:1 for
    stmt (3):2:6 ;
        expr (2):2:6 x
    expr (2):2:8 x
    expr (2):2:10 x
    stmt (3):2:12 ;
        expr (2):2:12 x
stmt (10):3:1 for
    stmt (4):3:6 var i
        expr (2):3:14 0
        type (1):3:10 (inferred)
    expr (6):3:17 (21)<
        expr (2):3:17 i
        expr (2):3:21 10
    expr (5):3:25 (1)++
        expr (2):3:25 i
    stmt (1):3:29 (nop)
